contracts/external/traders/CanonicalOrders.sol

```diff
@@ -0,0 +1,842 @@
+    limitations under the License.
+
+*/
+
+pragma solidity 0.5.7;
+pragma experimental ABIEncoderV2;
+
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import { Ownable } from "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import { IAutoTrader } from "../../protocol/interfaces/IAutoTrader.sol";
+import { ICallee } from "../../protocol/interfaces/ICallee.sol";
+import { Account } from "../../protocol/lib/Account.sol";
+import { Math } from "../../protocol/lib/Math.sol";
+import { Monetary } from "../../protocol/lib/Monetary.sol";
+import { Require } from "../../protocol/lib/Require.sol";
+import { Types } from "../../protocol/lib/Types.sol";
+import { OnlySolo } from "../helpers/OnlySolo.sol";
+import { TypedSignature } from "../lib/TypedSignature.sol";
+
+
+/**
+ * @title CanonicalOrders
+ * @author dYdX
+ *
+ * Allows for Canonical Orders to be used with dYdX
+ */
+contract CanonicalOrders is
+    Ownable,
+    OnlySolo,
+    IAutoTrader,
+    ICallee
+{
+    using Math for uint256;
+    using SafeMath for uint256;
+    using Types for Types.Par;
+    using Types for Types.Wei;
+
+    // ============ Constants ============
+
+    bytes32 constant private FILE = "CanonicalOrders";
+
+    // EIP191 header for EIP712 prefix
+    bytes2 constant private EIP191_HEADER = 0x1901;
+
+    // EIP712 Domain Name value
+    string constant private EIP712_DOMAIN_NAME = "CanonicalOrders";
+
+    // EIP712 Domain Version value
+    string constant private EIP712_DOMAIN_VERSION = "1.0";
+
+    // Hash of the EIP712 Domain Separator Schema
+    /* solium-disable-next-line indentation */
+    bytes32 constant private EIP712_DOMAIN_SEPARATOR_SCHEMA_HASH = keccak256(abi.encodePacked(
+        "EIP712Domain(",
+        "string name,",
+        "string version,",
+        "uint256 chainId,",
+        "address verifyingContract",
+        ")"
+    ));
+
+    // Hash of the EIP712 CanonicalOrder struct
+    /* solium-disable-next-line indentation */
+    bytes32 constant private EIP712_ORDER_STRUCT_SCHEMA_HASH = keccak256(abi.encodePacked(
+        "CanonicalOrder(",
+        "bytes32 flags,",
+        "uint256 baseMarket,",
+        "uint256 quoteMarket,",
+        "uint256 amount,",
+        "uint256 limitPrice,",
+        "uint256 triggerPrice,",
+        "uint256 limitFee,",
+        "address makerAccountOwner,",
+        "uint256 makerAccountNumber,",
+        "address taker,",
+        "uint256 expiration",
+        ")"
+    ));
+
+    // Number of bytes in an Order struct (plus orderInfo.price/fee/isNegativeFee)
+    uint256 constant private NUM_ORDER_BYTES = 448;
+
+    // Number of bytes in a typed signature
+    uint256 constant private NUM_SIGNATURE_BYTES = 66;
+
+    // The number of decimal places of precision in the price ratio of a triggerPrice
+    uint256 constant private PRICE_BASE = 10 ** 18;
+
+    // Bitmasks for the order.flag argument
+    bytes32 constant private IS_BUY_FLAG = bytes32(uint256(0xf));
```

Comment:
It does not for some reason unless the string is actually 32 bytes long. You could also cast to a bytes1 and then a bytes32, but then the bytes would be at the start of the bytes32 instead of the end